proposals/001-file-format-api.md

```diff
@@ -0,0 +1,608 @@
+`Audiobook` | application/audiobook+zip
+`AudiobookManifest` | application/audiobook+json
+`Avi` | video/x-msvideo
+`Binary` | application/octet-stream
+`Bmp` | image/bmp
+`Cbz` | application/vnd.comicbook+zip
+`Css` | text/css
+`DiViNa` | application/divina+zip
+`DiViNaManifest` | application/divina+json
+`Epub` | application/epub+zip
+`Gif` | image/gif
+`Gz` | application/gzip
+`JavaScript` | text/javascript
+`Jpeg` | image/jpeg
+`Html` | text/html
+`Opds1` | application/atom+xml;profile=opds-catalog
+`Opds1Entry` | application/atom+xml;type=entry;profile=opds-catalog
+`Opds2` | application/opds+json
+`Opds2Publication` | application/opds-publication+json
+`Json` | application/json
+`LcpProtectedAudiobook` | application/audiobook+lcp
+`LcpProtectedPdf` | application/pdf+lcp
+`LcpLicenseDocument` | application/vnd.readium.lcp.license.v1.0+json
+`LcpStatusDocument` | application/vnd.readium.license.status.v1.0+json
+`Mp3` | audio/mpeg
+`Mpeg` | video/mpeg
+`Ogg` | audio/ogg
+`Ogv` | video/ogg
+`Opus` | audio/opus
+`Otf` | font/otf
+`Pdf` | application/pdf
+`Png` | image/png
+`Svg` | image/svg+xml
+`Text` | text/plain
+`Tiff` | image/tiff
+`Ttf` | font/ttf
+`Wav` | audio/wav
+`WebmAudio` | audio/webm
+`WebmVideo` | video/webm
+`Webp` | image/webp
+`WebPub` | application/webpub+zip
+`WebPubManifest` | application/webpub+json
+`Woff` | font/woff
+`Woff2` | font/woff2
+`Xhtml` | application/xhtml+xml
+`Xml` | application/xml
+`Zip` | application/zip
+
+
+### `Format` Class
+
+Instead of using directly a media type, we use a `Format` data class to identify a file format for several reasons:
+
+* A format such as CBZ can be identified by several media types.
+* It's a convenient way to pair a media type and its name and file extension.
+
+#### Constructor
+
+Note: all of the constructor parameters are exposed as read-only public properties.
+
+* `Format(mediaType: MediaType, name: String, fileExtension: String)`
+  * `mediaType: MediaType`
+    * The canonical media type that identifies the best (most officially) this format.
+    * This will be stored in the reading app's database, for example.
+  * `name: String`
+    * A human readable name describing the format.
+    * Can be presented to the user.
+  * `fileExtension: String`
+    * The default file extension to use for this format.
+
+#### Helpers
+
+* `isManifest`, `isOpds`, `isHtml`, `isBitmap`...
+  * Aliases to `mediaType.isX`
+
+#### Constants
+
+* `defaultSniffers: List<Sniffer>`
+  * The default sniffers provided by Readium 2 to guess a `Format`.
+  * This list is writable, to allow reading apps to register additional sniffers globally, or remove sniffers.
+
+Formats used by Readium are represented as static constants (singletons) on `Format`, for convenience.
+Reading apps are welcome to extend the static constants with additional formats.
+
+Constant | Name | Extension | Media Type
+-------- | ---- | --------- | ----------
+`Audiobook` | Audiobook | audiobook | application/audiobook+zip
```

Comment:
@llemeurfr You might want to review the name of each format, especially for the LCP and RWPM formats.